TPP.Core/Commands/Definitions/BadgeCommands.cs

```diff
@@ -134,15 +141,150 @@ public async Task<CommandResult> SelectBadge(CommandContext context)
 
         public async Task<CommandResult> Pokedex(CommandContext context)
         {
-            User user = (await context.ParseArgs<Optional<User>>()).OrElse(context.Message.User);
+            (Optional<User> optionalUser, Optional<string> mode_, Optional<User> optionalUser2) =
+                await context.ParseArgs<Optional<User>, Optional<string>,Optional<User>>();
+            User user = optionalUser.IsPresent ? optionalUser.Value : context.Message.User;
             bool isSelf = user == context.Message.User;
-            int numUniqueSpecies = (await _badgeRepo.CountByUserPerSpecies(user.Id)).Count;
-            return new CommandResult
+
+            // Have to additionally check if the string is not empty, otherwise "!dex " would jump right into the else - branch.
+            //   Not sure if this is an actual issue when connected to twitch
+            if( !mode_.IsPresent || string.IsNullOrWhiteSpace( mode_.ToString() ) )
             {
-                Response = isSelf
-                    ? $"You have collected {numUniqueSpecies} distinct Pokémon badge(s)"
-                    : $"{user.Name} has collected {numUniqueSpecies} distinct Pokémon badge(s)"
-            };
+                int numUniqueSpecies = (await _badgeRepo.CountByUserPerSpecies(user.Id)).Count;
+                return new CommandResult
+                {
+                    Response = isSelf
+                        ? $"You have collected {numUniqueSpecies} distinct Pokémon badge(s)"
+                        : $"{user.Name} has collected {numUniqueSpecies} distinct Pokémon badge(s)"
+                };
+            }
+            else
+            {
+                string mode = mode_.ToString();
+                ImmutableSortedDictionary<PkmnSpecies, int> numBadgesPerSpecies =
+                    await _badgeRepo.CountByUserPerSpecies( user.Id );
+
+                if( mode.Equals( PokedexModeMissing ))
+                {
+                    IEnumerable<PkmnSpecies> missingList = PokedexData.Load().KnownSpecies.Except( numBadgesPerSpecies.Keys );
```

Comment:
`PokedexData.Load()` is an expensive call. please pass the already-loaded pokedex data into this class' constructor instead. See https://discord.com/channels/148079346685313034/436252785873387540/817502346145562676 and I can also assist you with that if needed

Reply:
Wrong usage of Singleton. Thats embarrasing

Reply:
lol, forget the last comment. That was just me being pepega. I passed the pokedexData - MemberVariable to BadgeCommands as you suggested.